pkg/thanos/rules.go

```diff
@@ -220,15 +223,37 @@ func (o *Operator) selectRules(t *monitoringv1.ThanosRuler, namespaces []string)
 	return rules, nil
 }
 
-func generateContent(promRule monitoringv1.PrometheusRuleSpec) (string, error) {
-
+func generateContent(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
 	content, err := yaml.Marshal(promRule)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal content")
 	}
+	errs := validateRule(promRule)
+	if len(errs) != 0 {
+		const m = "Invalid rule"
+		level.Debug(logger).Log("msg", m, "content", content)
+		for _, err := range errs {
+			level.Info(logger).Log("msg", m, "err", err)
+		}
+		return "", errors.New(m)
+	}
 	return string(content), nil
 }
 
+func validateRule(promRule monitoringv1.PrometheusRuleSpec) []error {
```

Comment:
Can we not duplicate code and reuse same function from `pkg/prometheus/rules.go` ?

Reply:
Thought about that, yes. Currently prometheus and thanos are decoupled packages and couldn't find a common package to factor out this validation logic.

if you think that thanos pkg could depend on prometheus pkg I can do that

Reply:
The validation logic should be the same in all 3 places (prometheus, thanos, and admission webhook packages), and introducing dependency will allow us to keep it that way in the long run. So yes, let's introduce it.